radixdlt/babylon-wallet-android
# [ABW-3438] - Rich AF installation link

Jira ticket: [ABW-3437](https://radixdlt.atlassian.net/browse/ABW-3438)

## Description
This PR resolves deferred deep links via AppsFlyer. This will allow new users to tap on a link that will redirect them to the Google Play store, but when the app is downloaded and opened, it will fetch specific information sent on the link query params.
For now, we are just logging this information. In future PRs we will deal with it.

## How to test
1. Delete every Radix Wallet app from your device (should work just by deleting Dev, but I've tested without any other)
2. Select any One Link on the [AppsFlyer dashboard](https://hq1.appsflyer.com/onelink) from test template. For example, you can select: 
```
https://radixdlt.onelink.me/WAKF/eun9mos3
```
3. Append the query parameters you would like to catch under `deep_link_value` or `deep_lin_sub1/10`. For example, it should be like:
```
https://radixdlt.onelink.me/WAKF/eun9mos3?deep_link_value=reviewPR&deep_link_sub1=thisWorks
```
4. Copy the link on your device (e.g. in Notes app), and tap on it. Your browser will open indicating `The app you are looking for is unavailable`. This is because Dev (debug) isn't available on the Google Play store.
5. Run the app from Android Studio to simulate an install.
6. Check the logs and filter by `Click event`. You should see something like the following:
```
did resolve deep link. Is deferred: true. 
Click event: {"campaign_id":"","af_sub3":"","match_type":"probabilistic","af_sub1":"",
"deep_link_value":"reviewPR","campaign":"","af_sub4":"","timestamp":
"2024-06-28T17:25:35.362","click_http_referrer":"","af_sub5":"","media_source":"","af_sub2":"",
"deep_link_sub1":"thisWorks","is_deferred":true}
```

You can see that both query parameters have been intercepted by the app after its installation.

> Note: You must have secrets.properties file at ../config/secrets/ path with appsFlyerDevKey defined.


[ABW-3437]: https://radixdlt.atlassian.net/browse/ABW-3437?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
